src/renderer/components/dialogs/PreviewTargetDialog.vue

```diff
@@ -0,0 +1,267 @@
+        <input
+          id="deployOption"
+          type="radio"
+          name="previewType"
+          :checked="isDeploy" 
+          @change="setPreviewType('deploy')"
+        />
+        <label for="previewType">Deploy Folder</label>
+
+        <br />
+
+        <input
+          id="urlOption"
+          type="radio"
+          name="previewType"
+          :checked="isURL"
+          @change="setPreviewType('url')"
+        />
+        <label for="previewType">Custom URL</label>
+
+        <br />
+
+        <input
+          id="urlInput" 
+          class="urlInput" 
+          :disabled="disableURL() === true" 
+          :value="previewURL" 
+          @input="onUrlInputChange()"
+        />
+      </div>
+
+      <div class="actions">
+        <button 
+          id="cancelBtn"
+          @click="onBtnCancelClick()"
+        >
+          Cancel
+        </button>
+        <button
+          id="confirmBtn"
+          :disabled="disableConfirm() === true"
+          @click="onBtnConfirmClick()"
+        >
+          Confirm
+        </button>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+import { mapState } from 'vuex';
+
+export default {
+  props: {
+    // Variables
+    visible: {
+      type:Boolean,
+      default: false
+    },
+
+    // Callbacks
+    onCancel: { 
+      type: Function,
+      default: () => {}
+    },
+    onConfirm: {
+      type: Function,
+      default: () => {}
+    }
+  },
+
+  /**
+   * Data object
+   */
+  data: function () {
+    return {
+      previewType: undefined
+    };
+  },
+
+  computed: {
+    ...mapState({
+      /**
+       * Returns the last known preview target from storage.
+       */
+      previewTarget: function(state) {
+        return (!state.gamePreview || !state.gamePreview.previewTarget) ? undefined : state.gamePreview.previewTarget;
+      },
+
+      /**
+       * Returns the last known preview URL from storage.
+       */
+      previewURL: function(state) {
+        if (this.isDeploy || this.previewType === 'deploy') {
+          return '';
+        }
+        return (!state.gamePreview || !state.gamePreview.previewURL) ? '' : state.gamePreview.previewURL;
+      },
+
+      /**
+       * Returns whether deploy was the last known preview target.
+       */
+      isDeploy: function(state) {
+        return state.gamePreview && state.gamePreview.previewTarget && state.gamePreview.previewTarget === 'deploy';
+      },
+
+      /**
+       * Returns whether url was the last known preview target.
+       */
+      isURL: function(state) {
+        return state.gamePreview && state.gamePreview.previewTarget && state.gamePreview.previewTarget === 'url';
+      }
+    })
+  },
+
+  /**
+   * When this component is mounted, set some states.
+   */
+  mounted: function() {
+    this.previewType = this.previewTarget;
+  },
+
+  methods: {
+    /**
+     * Whether or not the URL input field should be disabled.
+     */
+    disableURL: function() {
```

Comment:
This should probably be moved to a computed property. Computed properties benefit from some caching so they're a little faster.